vlog4j-parser/src/test/java/org/semanticweb/vlog4j/syntax/parser/EntityTest.java

```diff
@@ -0,0 +1,190 @@
+		PositiveLiteral headAtom1 = Expressions.makePositiveLiteral("q", x, z);
+		Rule rule1 = Expressions.makeRule(headAtom1, atom1);
+		assertEquals(rule1, RuleParser.parseRule(rule1.toString()));
+	}
+
+	@Test
+	public void datatypeDoubleConstantToStringRoundTripTest() throws ParsingException {
+		String shortDoubleConstant = "12.345E67";
+		assertEquals(shortDoubleConstant,
+				RuleParser.parseFact("p(\"12.345E67\"^^<http://www.w3.org/2001/XMLSchema#double>).").getArguments()
+						.get(0).toString());
+		assertEquals(shortDoubleConstant, RuleParser.parseFact("p(12.345E67).").getArguments().get(0).toString());
+	}
+
+	@Test
+	public void datatypeFloatConstantToStringRoundTripTest() throws ParsingException {
+		String floatConstant = "\"0.5\"^^<http://www.w3.org/2001/XMLSchema#float>";
+		assertEquals(floatConstant, RuleParser.parseFact("p(\"0.5\"^^<http://www.w3.org/2001/XMLSchema#float>).")
+				.getArguments().get(0).toString());
+	}
+
+	@Test
+	public void datatypeStringConstantToStringRoundTripTest() throws ParsingException {
+		String shortStringConstant = "\"data\"";
+		assertEquals(shortStringConstant, RuleParser
+				.parseFact("p(\"data\"^^<http://www.w3.org/2001/XMLSchema#string>).").getArguments().get(0).toString());
+		assertEquals(shortStringConstant, RuleParser.parseFact("p(\"data\").").getArguments().get(0).toString());
+	}
+
+	@Test
+	public void datatypeIntegerConstantToStringRoundTripTest() throws ParsingException {
+		String shortIntegerConstant = "1";
+		assertEquals(shortIntegerConstant, RuleParser.parseFact("p(\"1\"^^<http://www.w3.org/2001/XMLSchema#integer>).")
+				.getArguments().get(0).toString());
+		assertEquals(shortIntegerConstant, RuleParser.parseFact("p(1).").getArguments().get(0).toString());
+	}
+
+	@Test
+	public void datatypeDecimalToStringRoundTripTest() throws ParsingException {
+		String shortDecimalConstant = "0.23";
+		assertEquals(shortDecimalConstant,
+				RuleParser.parseFact("p(\"0.23\"^^<http://www.w3.org/2001/XMLSchema#decimal>).").getArguments().get(0)
+						.toString());
+		assertEquals(shortDecimalConstant, RuleParser.parseFact("p(0.23).").getArguments().get(0).toString());
+	}
+
+	@Test
+	public void sparqlDataSourceDeclarationToStringParsingTest() throws ParsingException, IOException {
+		KnowledgeBase kb = new KnowledgeBase();
+		Predicate predicate1 = Expressions.makePredicate("p", 3);
+		SparqlQueryResultDataSource dataSource = new SparqlQueryResultDataSource(new URL("https://example.org/sparql"),
+				"var", "?var wdt:P31 wd:Q5 .");
+		DataSourceDeclaration dataSourceDeclaration1 = new DataSourceDeclarationImpl(predicate1, dataSource);
+		RuleParser.parseInto(kb, dataSourceDeclaration1.toString());
+		assertEquals(dataSourceDeclaration1, kb.getDataSourceDeclarations().get(0));
+	}
+
+	@Test
+	public void rdfDataSourceDeclarationToStringParsingTest() throws ParsingException, IOException {
+		KnowledgeBase kb = new KnowledgeBase();
+		String INPUT_FOLDER = "src/test/data/input/";
```

Comment:
this is a local variable, it should not have a constant (class public static final fields) naming convention.